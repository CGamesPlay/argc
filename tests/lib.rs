use insta::assert_snapshot;

#[macro_export]
macro_rules! assert_argc {
    (
        $source:expr,
        $args:expr
    ) => {
        let (stdout, stderr) = argc::run($source, $args).unwrap();
        assert_snapshot!(stderr.unwrap_or_default());
        assert_snapshot!(stdout.unwrap_or_default());
    };
}

#[test]
fn test_git() {
    assert_argc!(include_str!("git.sh"), &["git", "-h"]);
}

#[test]
fn test_git_add() {
    assert_argc!(include_str!("git.sh"), &["git", "add", "-h"]);
}

#[test]
fn test_git_remote() {
    assert_argc!(include_str!("git.sh"), &["git", "push", "-h"]);
}

#[test]
fn test_git_log() {
    assert_argc!(include_str!("git.sh"), &["git", "log", "-h"]);
}
