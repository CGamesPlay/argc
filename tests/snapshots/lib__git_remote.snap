---
source: tests/lib.rs
assertion_line: 27
expression: stderr.unwrap_or_default()

---
git-push 
Update remote refs along with associated objects

USAGE:
    git push <repository> <refspec>...

ARGS:
    <repository>    The "remote" repository that is destination of a push operation.
    <refspec>...    Specify what destination ref to update with what source object.

OPTIONS:
    -h, --help    Print help information

